components/dashboard/src/teams/variables/NamedOrganizationEnvvarItem.tsx

```diff
@@ -0,0 +1,163 @@
+import { OrganizationRemoveEnvvarModal } from "./OrganizationRemoveEnvvarModal";
+import { InputField } from "../../components/forms/InputField";
+import { ReactComponent as Stack } from "../../icons/Repository.svg";
+import { Button } from "@podkit/buttons/Button";
+import { useCreateOrganizationEnvironmentVariable } from "../../data/organizations/org-envvar-queries";
+import Modal, { ModalBody, ModalFooter, ModalFooterAlert, ModalHeader } from "../../components/Modal";
+import { TextInputField } from "../../components/forms/TextInputField";
+import { useToast } from "../../components/toasts/Toasts";
+import { LoadingButton } from "@podkit/buttons/LoadingButton";
+
+type Props = {
+    disabled?: boolean;
+    organizationId: string;
+    name: string;
+    variable: OrganizationEnvironmentVariable | undefined;
+};
+export const NamedOrganizationEnvvarItem = ({ disabled, organizationId, name, variable }: Props) => {
+    const [showRemoveModal, setShowRemoveModal] = useState<boolean>(false);
+    const [showAddModal, setShowAddModal] = useState<boolean>(false);
+
+    const value = variable ? "*****" : "not set";
+
+    return (
+        <>
+            {variable && showRemoveModal && (
+                <OrganizationRemoveEnvvarModal
+                    variable={variable}
+                    organizationId={organizationId}
+                    onClose={() => setShowRemoveModal(false)}
+                />
+            )}
+
+            {showAddModal && (
+                <AddOrgEnvironmentVariableModal
+                    organizationId={organizationId}
+                    staticName={name}
+                    onClose={() => setShowAddModal(false)}
+                />
+            )}
+
+            <InputField disabled={disabled} className="w-full max-w-lg">
+                <div className="flex flex-col bg-gray-50 dark:bg-gray-800 p-3 rounded-lg">
+                    <div className="flex items-center justify-between">
+                        <div className="flex-1 flex items-center overflow-hidden h-8" title={value}>
+                            <span className="w-5 h-5 mr-1">
+                                <Stack />
+                            </span>
+                            <span className="truncate font-medium text-gray-700 dark:text-gray-200">{name}</span>
+                        </div>
+                        {!disabled && !variable && (
+                            <Button variant="link" onClick={() => setShowAddModal(true)}>
+                                Add
+                            </Button>
+                        )}
+                        {!disabled && variable && (
+                            <Button variant="link" onClick={() => setShowRemoveModal(true)}>
+                                Delete
+                            </Button>
+                        )}
+                    </div>
+                    <div className="mx-6 text-gray-400 dark:text-gray-500 truncate">
```

Comment:
If you decide to accept my above suggestion, this goes along with it (so that this is horizontally aligned with the variable name):
```suggestion
                    <div className="mx-7 text-gray-400 dark:text-gray-500 truncate">
```